### PR TITLE
Pdct 981/admin api if language idempotent language is removed from document

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

When updating a document, the user language was being correctly passed to the API - but if the language was idempotent the API was setting the user language as None in the database. 

This PR fixes that bug 🐛 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

Have updated update document tests to make sure we are only testing one thing edge case test, as the edge case was not being tested as we were always changing all of the fields in each test. I think we need to review the slug tests however, as I'm not confident that they're working as expected anymore ([ticket for doing](https://linear.app/climate-policy-radar/issue/PDCT-983/review-and-update-slug-tests-in-admin-service)).

## Reviewer Checklist

- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/repository/document.py [hunk 1]
```diff
@@ -228,15 +228,9 @@ def update(db: Session, import_id: str, document: DocumentWriteDTO) -> bool:
...
-        .filter(
-            PhysicalDocumentLanguage.document_id == original_fd.physical_document_id
-        )
-        .filter(PhysicalDocumentLanguage.source == LanguageSource.USER)
-        .one_or_none()
-    )
-    new_language = _get_new_language(db, new_values, existing_language)
+    existing_language = _get_existing_language(db, original_fd)
+    new_language = _get_requested_language(db, new_values)
+    lang_idempotent = _is_language_idempotent(existing_language, new_language)
```
Comment (diversemix) [inline app/repository/document.py:233]: would this read better if it were called `has_language_changed` ?

**Review by diversemix — APPROVED** [review]
Great nice tests too!

File: app/repository/document.py [hunk 2]
```diff
@@ -303,24 +303,49 @@ def update(db: Session, import_id: str, document: DocumentWriteDTO) -> bool:
...
+
+    new_language = (
+        db.query(Language)
+        .filter(Language.name == new_values["user_language_name"])
+        .one()
+    )
+    return new_language
+
+
+def _is_language_idempotent(
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply app/repository/document.py]: Have updated
